core/io/primitives.h

```diff
@@ -45,6 +52,9 @@ namespace Gadgetron::Core::IO {
         return val;
     }
 
+    template<class TObjectType>
+    void read_objects_from_stream(std::istream &stream, Gadgetron::hoNDArray<TObjectType> &array);
```

Comment:
I am wondering if we should opt for the slightly more verbose `read_array_of_objects` (you can skip stream in the name to shorten).  Or even `read_many` to link better with `read` above. It could just be `read`, but I see how you are trying to maneuver around some issues using the different name. I think I would suggest `read_many` and `write_many`